src/Bicep.LangServer/Deploy/DeploymentOperationsCache.cs

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Collections.Concurrent;
+using Azure.ResourceManager;
+using Azure.ResourceManager.Resources;
+
+namespace Bicep.LanguageServer.Deploy
+{
+    public class DeploymentOperationsCache : IDeploymentOperationsCache
+    {
+        private readonly ConcurrentDictionary<string, ArmOperation<ArmDeploymentResource>> deploymentOperationsCache = new ConcurrentDictionary<string, ArmOperation<ArmDeploymentResource>>();
+
+        public void CacheDeploymentOperation(string deploymentId, ArmOperation<ArmDeploymentResource> armOperation)
+        {
+            deploymentOperationsCache.TryAdd(deploymentId, armOperation);
+        }
+
+        public ArmOperation<ArmDeploymentResource>? GetDeploymentOperation(string deploymentId)
```

Comment:
Get* isn't a good name for a destructive operation.  Maybe FindAndRemoveDeploymentOperation or just RemoveDeploymentOperation?

Reply:
Sure, updated here:
https://github.com/Azure/bicep/commit/4dffe33c8366498862044f6792cef47f151cb0c3